WordPress/Classes/Models/PublicizeService+Lookup.swift

```diff
@@ -0,0 +1,31 @@
+extension PublicizeService {
+    /// Finds a cached `PublicizeService` matching the specified service name.
+    ///
+    /// - Parameter name: The name of the service. This is the `serviceID` attribute for a `PublicizeService` object.
+    ///
+    /// - Returns: The requested `PublicizeService` or nil.
+    ///
+    static func lookupPublicizeServiceNamed(_ name: String, in context: NSManagedObjectContext) throws -> PublicizeService? {
+        let request = NSFetchRequest<PublicizeService>(entityName: PublicizeService.classNameWithoutNamespaces())
+        request.predicate = NSPredicate(format: "serviceID = %@", name)
+        return try context.fetch(request).first
```

Comment:
Not sure I called this out before, but one great benefit of all this work is how much more Swifty the code is getting by leveraging `throws` in many more places as opposed to swallowing errors or return `.none` as a sign of error.